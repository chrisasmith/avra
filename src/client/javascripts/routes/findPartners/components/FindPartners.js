import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { FormattedMessage } from 'react-intl'
import * as findPartnersActions from '../../../reducers/pages/findPartnersActions'
import FilterBar from '../../../components/Filter/FilterBar'
import { createLoadingSelector, createErrorMessageSelector } from '../../../reducers/selectors'
import styles from './styles/FindPartners.scss'
import Tabs from './Tabs'
import DASHBOARD_DATA from './Data'

const propTypes = {
  getFindPartnersData: PropTypes.func.isRequired,
  findPartnersPageDataSuccess: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired,
  params: PropTypes.object.isRequired,
  filter: PropTypes.object.isRequired,
  children: PropTypes.element,
  data: PropTypes.object,
}

export const PROPERTY_TYPE = {
  SPORT: 'sport',
  LEAGUE: 'league',
  TEAM: 'team',
}

class FindPartners extends Component {
  constructor(props, context) {
    super(props, context)

    this.onFilterChange = this.onFilterChange.bind(this)

    this.state = {
      filter: { ...props.filter, ages: { ...props.filter.ages } },
    }
  }

  componentDidMount() {
    this.props.findPartnersPageDataSuccess(DASHBOARD_DATA)
  }

  componentWillReceiveProps(next) {
    if ((next.params.industry_id !== this.props.params.industry_id) ||
      (next.params.subsection_id !== this.props.params.subsection_id)) {
      this.refresh()
    }
  }

  refresh() {
    this.props.getFindPartnersData()
  }

  onFilterChange(filter) {
    this.setState({ filter }, () => this.props.getFindPartnersData(filter))
  }

  render() {
    return (
      <div className={styles.columns}>
        <div>
          <FilterBar
            onFilterChange={this.onFilterChange}
          />
        </div>
        <div>
          <div className={styles.title}>
            <h4>
              <FormattedMessage
                id="find_partners.title"
                defaultMessage="Find Partners"
              />
            </h4>
          </div>
          <div className={styles.tabs}>
            <Tabs location={this.props.location} industries={this.props.data ? this.props.data.industries : null} />
          </div>
          {this.props.children}
        </div>
      </div>
    )
  }
}

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([findPartnersActions.GET_FIND_PARTNERS_PAGE.SELF])
  const errorSelector = createErrorMessageSelector([findPartnersActions.GET_FIND_PARTNERS_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.findPartners.data

  return { ...state, filter, isFetching: loadingSelector(state), errorMessage: errorSelector(state), data }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ ...findPartnersActions }, dispatch)
}

FindPartners.propTypes = propTypes
export default connect(mapStateToProps, mapDispatchToProps)(FindPartners)
