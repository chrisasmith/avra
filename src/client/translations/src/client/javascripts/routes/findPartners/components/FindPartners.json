[
  {
    "id": "find_partners.title",
    "defaultMessage": "Find Partners"
  }
]