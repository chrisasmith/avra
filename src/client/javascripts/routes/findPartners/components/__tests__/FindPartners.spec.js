import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallowToJson } from 'enzyme-to-json'
import FindPartners from '../FindPartners'

const mockStore = configureMockStore([thunk])

let store

const getFindPartnersPage = s => shallow(<FindPartners store={mockStore(s)} />).shallow()

describe('Find Partners Page', () => {
  beforeEach(() => {
    store = {
      location: { },
      params: { industry_id: null, subsection_id: null },
      loading: {

      },
      error: {

      },
      filters: {
        filter: {},
      },
      pages: {
        findPartners: {
          data: null,
        },
      },
    }
  })

  it('should show loading', () => {
    store.loading.GET_FIND_PARTNERS_PAGE = true

    const wrapper = getFindPartnersPage(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should fire refresh on location industry id change', () => {
    const getData = jest.fn()
    const mock = mockStore(store)
    const wrapper = shallow(<FindPartners store={mock} getFindPartnersData={getData} />).shallow()

    mock.clearActions()
    wrapper.instance().componentWillReceiveProps({ params: { industry_id: 1, subsection_id: null } })
    expect(mock.getActions()).toEqual([{ request: {}, type: 'GET_FIND_PARTNERS_PAGE_REQUEST' }])
  })

  it('should fire refresh on location industry child id change', () => {
    store.params.industry_id = 1
    const getData = jest.fn()
    const mock = mockStore(store)
    const wrapper = shallow(<FindPartners store={mock} getFindPartnersData={getData} />).shallow()

    mock.clearActions()
    wrapper.instance().componentWillReceiveProps({ params: { industry_id: 1, subsection_id: 2 } })
    expect(mock.getActions()).toEqual([{ request: {}, type: 'GET_FIND_PARTNERS_PAGE_REQUEST' }])
  })

  it('should show error', () => {
    store.error.GET_FIND_PARTNERS_PAGE = 'bad stuff happened'

    const wrapper = getFindPartnersPage(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
