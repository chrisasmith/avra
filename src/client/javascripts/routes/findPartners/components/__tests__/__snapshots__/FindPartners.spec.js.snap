// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Partners Page should show error 1`] = `
<div
  className="columns"
>
  <div>
    <Connect(FilterBar)
      onFilterChange={[Function]}
    />
  </div>
  <div>
    <div
      className="title"
    >
      <h4>
        Find Partners
      </h4>
    </div>
    <div
      className="tabs"
    >
      <Tabs
        industries={null}
        location={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`Find Partners Page should show loading 1`] = `
<div
  className="columns"
>
  <div>
    <Connect(FilterBar)
      onFilterChange={[Function]}
    />
  </div>
  <div>
    <div
      className="title"
    >
      <h4>
        Find Partners
      </h4>
    </div>
    <div
      className="tabs"
    >
      <Tabs
        industries={null}
        location={Object {}}
      />
    </div>
  </div>
</div>
`;
